### ISSUE TITLE
Math formula documentation for predict_partial_hazard with categorical features

### ISSUE THREAD

**Issue by tle4336** [CamDavidsonPilon/lifelines#1645]
Hi everyone, does anyone happen to know the formula that is used in `predict_partial_hazard` function of the class `CoxPHFitter` when the features have some categorical variables, each of which might have at least `3` values (e.g. IDs, day of week)?

**Comment by tle4336** [comment]
Could anyone please help with the above question?

**Comment by CamDavidsonPilon** [comment]
**reading the code**, categorical inputs are transformed into one-hot columns, and the mean of that column from the training set is subtracted, then betas are applied.

**Comment by tle4336** [comment]
Thank you very much for your help with my question, @CamDavidsonPilon. I have a quick clarification question: is the mean of categorical-input column **the same** as the mean given from the CphFitter()._norm_mean? For numerical-input columns, these two are the same, but I just want to ensure it remains that way for **categorical**.

**Comment by CamDavidsonPilon** [comment]
1. Yes,
2. I don't understand your question

**Comment by tle4336** [comment]
@CamDavidsonPilon Thank you very much for your quick reply. Let me rephrase question 2 by a concrete example: let's say we have `student ID` as one of the categorical-input column, with ID is an integer between 20 and 40 (inclusive). It means that we would have 20 one-hot encoding columns `x_21 to x_40`, where each of these columns would have their means computed by CoxPHFitter. Now, let' say the input contains `student ID = 21`  for **inference**. Then in the calculation of `partial hazard` term, we must have `(1 - mean of column x_21) * beta_{x21} + (0 - mean of column of x_22) * beta_{x22} +.... + (0 - mean of column of x_40) * beta_{x40} + [other terms associated with other predictors]` --- Is this really the case?

(https://web.archive.org/web/20070630025831/https://www.stat.nus.edu.sg/%7Estachenz/ST3242Notes3.pdf --- From **page 2** of this slide, without **de-meaning**, we won't have this sum: `(0 - mean of column of x_22) * beta_{x22} +.... + (0 - mean of column of x_40) * beta_{x40}.` But due to de-mean, we would have them? )

**Comment by CamDavidsonPilon** [comment]
>  do we actually have this sum: (1 - mean of column x_21) * beta_{x21} + (0 - mean of column of x_22) * beta_{x22} +.... + (0 - mean of column of x_40) * beta_{x40} + [other terms associated with other predictors] ?

We do, yes. 

Authors may choose the include demeaning in their formulas or not, but implementations must choose. Demeaning typically leads to better numerical stability, so lifelines demeans. Demeaning isn't that important, either: the output of the predict_partial_hazard is a meaningless number, only good for ranking / ratios, and a mean / demeaned prediction doesn't effect this.
